internal/hive/data.go

```diff
@@ -0,0 +1,62 @@
+// Package hive contains shared types for hive.
+package hive
+
+import (
+	"strconv"
+	"time"
+)
+
+// TestSuiteID identifies a test suite context.
+type TestSuiteID uint32
+
+func (tsID TestSuiteID) String() string {
+	return strconv.Itoa(int(tsID))
+}
+
+// TestID identifies a test case context.
+type TestID uint32
+
+func (tsID TestID) String() string {
+	return strconv.Itoa(int(tsID))
+}
+
+// TestSuite is a single run of a simulator, a collection of testcases.
+type TestSuite struct {
+	ID          TestSuiteID          `json:"id"`
+	Name        string               `json:"name"`
+	Description string               `json:"description"`
+	TestCases   map[TestID]*TestCase `json:"testCases"`
+	// the log-file pertaining to the simulator. (may encompass more than just one TestSuite)
+	SimulatorLog string `json:"simLog"`
+}
+
+// TestCase represents a single test case in a test suite.
+type TestCase struct {
+	ID            TestID                 `json:"id"`          // Test case reference number.
+	Name          string                 `json:"name"`        // Test case short name.
+	Description   string                 `json:"description"` // Test case long description in MD.
+	Start         time.Time              `json:"start"`
+	End           time.Time              `json:"end"`
+	SummaryResult TestResult             `json:"summaryResult"` // The result of the whole test case.
+	ClientResults map[string]*TestResult `json:"clientResults"` // Client specific results, if this test case supports this concept. Not all test cases will identify a specific client as a test failure reason.
+	ClientInfo    map[string]*ClientInfo `json:"clientInfo"`    // Info about each client.
+	PseudoInfo    map[string]*ClientInfo `json:"-"`             // registry of participating pseudos maintained for client maintenance and not part of the result database
+}
+
+// TestResult is the payload submitted to the EndTest endpoint.
+type TestResult struct {
+	Pass    bool   `json:"pass"`
+	Details string `json:"details"`
+}
+
+// ClientInfo describes a client that participated in a test case.
+type ClientInfo struct {
+	ID              string    `json:"id"`
+	IP              string    `json:"ip"`
```

Comment:
Maybe this should rather be a `map[string]net.IP` where key is the network name and value is the ip addr of the client on that network considering a client could be connected to several networks?

Reply:
We could change that later, but for now, we can't make any breaking changes to the API. It still needs to be compatible with simulators/common so the simulations keep working.